operator/cmd/mesh/profile-common.go

```diff
@@ -214,6 +221,50 @@ func genIOPSFromProfile(profileOrPath, fileOverlayYAML, setOverlayYAML string, s
 	return util.ToYAMLWithJSONPB(finalIOPS), finalIOPS, nil
 }
 
+// translateK8SfromValueToIOP use reverse translation to convert k8s settings defined in values API to IOP API.
+func translateK8SfromValueToIOP(userOverlayYaml string, l *Logger) (string, error) {
+	mvs := version.OperatorBinaryVersion.MinorVersion
+	t, err := translate.NewReverseTranslator(mvs)
+	if err != nil {
+		return "", fmt.Errorf("error creating values.yaml translator: %s", err)
+	}
+	valuesOverlay, err := tpath.GetConfigSubtree(userOverlayYaml, "spec.values")
+	if err != nil {
+		scope.Debugf("no spec.values section from userOverlayYaml %v", err)
+		return "", nil
+	}
+	var valuesOverlayTree = make(map[string]interface{})
+	err = yaml.Unmarshal([]byte(valuesOverlay), &valuesOverlayTree)
+	if err != nil {
+		return "", fmt.Errorf("error unmarshalling values overlay yaml into untype tree %v", err)
+	}
+	iopSpecTree := make(map[string]interface{})
+	if err = t.TranslateK8S(valuesOverlayTree, iopSpecTree); err != nil {
+		return "", err
+	}
+	warning, err := t.IssueWarningForGatewayK8SSettings(valuesOverlay)
+	if err != nil {
+		return "", fmt.Errorf("error handling values gateway k8s settings: %v", err)
+	}
+	if warning != "" {
+		l.logAndPrint(warning)
```

Comment:
this should probably be a hard error?

Reply:
so you mean we should fail the command with warning instead of proceed with warning?

Reply:
right, because i think proceeding may lead to unexpected config being applied.

Reply:
done